examples/dnsdoctor/dnsdoctor.lua

```diff
@@ -0,0 +1,77 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- DNS Doctoring : Rewrite DNS type A record to a private address for local clients
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local function nop() end
+
+local function get_domain(skb, off)
+	local name = ""
+	local i = 0
+
+	repeat
+		local d = skb:getbyte(off + i)
+		i = i + 1
+	until d == 0
+
+	return name .. skb:getstring(off, i-1), i
+end
+
+local function dnsdoctor_tg(skb, par, userdata)
+	local sep = userdata:find(" ")
```

Comment:
you can encode strings prepended with its size.. take a look at https://www.lua.org/manual/5.4/manual.html#6.4.2:

```
s[n]: a string preceded by its length coded as an unsigned integer with n bytes (default is a size_t)
```

then, you just  need to unpack firstly

```
T: a size_t (native size)
```

Reply:
I used this because, `lunatik.com` is actually encoded as `\x07lunatik\x03com` in the lua program, this directly matches that of the DNS answer format that is `<len of string>string`, so I used a seperator to seperate this and the IP addresses.

Reply:
Unpacking this into `lunatik` and `com` will need additional step to search for each string in the DNS answer, whereas `x07lunatik\x03com` directly matches the string in DNS packet.

Reply:
sorry.. I think I'm missing something..

```
> domain = "\x07lunatik\x03com"
> userarg = string.pack("sI8", "\x07lunatik\x03com", 0xFF)
> string.unpack("sI8", userarg)
lunatikcom	255	29
```

wouldn't this be enough?

Reply:
Yes, this can be done, through your approach, I identified a bug in my code too. Earlier, if the binary string returned by pack had `\x00` in the beginning part, the userdata was not passed correctly to the kernel lua program, since I was using `lua_pushstring`, now I changed it to `lua_pushlstring(info->userdata, LUAXTABLE_USERDATA_SIZE)`

Reply:
Reference : https://github.com/luainkernel/lunatik/pull/153/files#diff-292f6a11c6f52f91544b0b47343e04b5d6516316ca5a075844659348899709ffR59